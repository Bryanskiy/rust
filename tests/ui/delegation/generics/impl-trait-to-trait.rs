#![feature(fn_delegation)]
#![allow(incomplete_features)]

mod bounds {
    trait Trait0 {}

    trait Trait1<T> {
        fn foo<U>(&self)
        where
            T: Trait0,
            U: Trait0,
            Self: Trait0,
            //~^ ERROR the trait bound `S: Trait0` is not satisfied
        {
        }
    }

    struct F;
    impl<T> Trait1<T> for F {}

    struct S(F);

    impl<T> Trait1<T> for S {
        reuse Trait1::<T>::foo { &self.0 }
        //~^ ERROR the trait bound `F: Trait0` is not satisfied
    }
}

fn main() {}
