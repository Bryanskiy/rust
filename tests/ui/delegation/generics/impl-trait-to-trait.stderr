error[E0277]: the trait bound `S: Trait0` is not satisfied
  --> $DIR/impl-trait-to-trait.rs:12:19
   |
LL |             Self: Trait0,
   |                   ^^^^^^ the trait `Trait0` is not implemented for `S`
   |
help: this trait has no implementations, consider adding one
  --> $DIR/impl-trait-to-trait.rs:5:5
   |
LL |     trait Trait0 {}
   |     ^^^^^^^^^^^^
   = help: see issue #48214
help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
   |
LL + #![feature(trivial_bounds)]
   |

error[E0277]: the trait bound `F: Trait0` is not satisfied
  --> $DIR/impl-trait-to-trait.rs:24:32
   |
LL |         reuse Trait1::<T>::foo { &self.0 }
   |                            --- ^^-------^^
   |                            |   | |
   |                            |   | this tail expression is of type `&F`
   |                            |   the trait `Trait0` is not implemented for `F`
   |                            required by a bound introduced by this call
   |
help: this trait has no implementations, consider adding one
  --> $DIR/impl-trait-to-trait.rs:5:5
   |
LL |     trait Trait0 {}
   |     ^^^^^^^^^^^^
note: required by a bound in `Trait1::foo`
  --> $DIR/impl-trait-to-trait.rs:12:19
   |
LL |         fn foo<U>(&self)
   |            --- required by a bound in this associated function
...
LL |             Self: Trait0,
   |                   ^^^^^^ required by this bound in `Trait1::foo`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
